src/core/Akka.Cluster.Tests/ClusterSpec.cs

```diff
@@ -192,6 +194,76 @@ public void A_cluster_must_complete_LeaveAsync_task_upon_being_removed()
             _cluster.LeaveAsync().IsCompleted.Should().BeTrue();
         }
 
+        [Fact(Skip = "This behavior not yet supported.")]
```

Comment:
Because wanted to discuss before 😄 

Right now `Leave()` and `LeaveAsync()` do not share state with each other.

`LeaveAsync()` memoizes the `_leaveTask`, therefore two subsequent calls of `LeaveAsync()` won't result in double sending of the leave request.

But `Leave()` doesn't take the `_leaveTask` into account. So, if cluster have been left by the `Leave()` method (or is in state of leaving) , `LeaveAsync()` doesn't know about it and will be waiting for the **MemberRemoved** event.

Here is how we can change the `Leave()` method to support that:

```c#
public void Leave(Address address)
{
    if (address == SelfAddress)
    {
        LeaveSelf();
    }
    else
        ClusterCore.Tell(new ClusterUserAction.Leave(FillLocal(address)));
}
```
I just wanted to check before, to be sure that it won't make any negative effect on anything.
Plus the self address comparison looks questionable.